sql/job_api.sql

```diff
@@ -23,9 +24,10 @@ CREATE OR REPLACE FUNCTION @extschema@.alter_job(
     scheduled BOOL = NULL,
     config JSONB = NULL,
     next_start TIMESTAMPTZ = NULL,
-    if_exists BOOL = FALSE
+    if_exists BOOL = FALSE, 
+    check_config REGPROC = NULL
 )
-RETURNS TABLE (job_id INTEGER, schedule_interval INTERVAL, max_runtime INTERVAL, max_retries INTEGER, retry_period INTERVAL, scheduled BOOL, config JSONB, next_start TIMESTAMPTZ)
+RETURNS TABLE (job_id INTEGER, schedule_interval INTERVAL, max_runtime INTERVAL, max_retries INTEGER, retry_period INTERVAL, scheduled BOOL, config JSONB, next_start TIMESTAMPTZ, check_config TEXT)
```

Comment:
Hmm... why `TEXT` and not `REGPROC`?

Reply:
I think there is one benefit here in returning `TEXT`. So alter_job returns either the new value of a field it updates, or the previous value of the field if it does not update it. In the case of the check function, the fields that the job stores are the check schema and the check name, not the Oid. 
A user is allowed to drop the check function they registered with add_job. Afterwards, they may run `alter_job` without updating the check function field. If we return REGPROC, then in order to find the previous value (and to be consistent with what `alter_job` returns), we will search for the oid of the dropped function, but we will not find anything, and won't be able to return the name of the previously registered function. Whereas if we return text, we can still return that information.  (I hope I am making sense!)

Reply:
Sounds good. Let's keep the text type.